vnext/Shared/OInstance.cpp

```diff
@@ -458,13 +479,35 @@ void InstanceImpl::loadBundleInternal(std::string &&jsBundleRelativePath, bool s
           m_devSettings->platformName,
           /*dev*/ true,
           /*hot*/ false,
-          m_devSettings->inlineSourceMap);
-
-      // Remote debug executor loads script from a Uri, rather than taking the actual bundle string
-      m_innerInstance->loadScriptFromString(
-          std::make_unique<const JSBigStdString>(m_devSettings->useWebDebugger ? bundleUrl : jsBundleString),
-          bundleUrl,
-          synchronously);
+          m_devSettings->inlineSourceMap,
+          hermesBytecodeVersion);
+
+      // This code is based on the HBC Bundle integration on Android
+      // Ref:
+      // https://github.com/facebook/react-native/blob/24d91268b64c7abbd4b26547ffcc663dc90ec5e7/ReactAndroid/src/main/jni/react/jni/CatalystInstanceImpl.cpp#L231
+      if (isHBCBundle(jsBundleString.c_str())) {
```

Comment:
this copies the string, is this necessary?

Reply:
`string_view` is often recommended for string params for this reason. A c string can be adapted to it without a copy (but it does do a length check).

But also a string_view cannot be safely converted back to c string because there are not guarantees around null terminator. A lot of the time I end up finding plain old C strings to be the lowest common denominator if you don't need to deal with ownership of them.

Reply:
Thanks for this .. I've updated the code to avoid copy/length-check by directly accessing the string buffer.. Please let me know what you think !

  // Note:: Directly access the pointer to avoid copy/length-check. It matters as this string contains the bundle which can be potentially huge.
  // https://herbsutter.com/2008/04/07/cringe-not-vectors-are-guaranteed-to-be-contiguous/#comment-483
  BundleHeader *header = reinterpret_cast<BundleHeader *>(const_cast<char *>(&bundle[0]));